examples/tests/sdk_tests/fs/json.main.w

```diff
@@ -0,0 +1,46 @@
+bring fs;
+bring regex;
+
+let filename = "test-json.json";
+let data = Json {
+    "foo": "bar", 
+    "arr": [1, 2, 3, "test", { "foo": "bar" }]
+};
+
+try {
+    fs.writeFile(filename, "invalid content");
+    fs.readJson(filename);
+} catch e {
+    assert(regex.match("^Unexpected token", e) == true);
+}
+
+fs.writeJson(filename, data);
+assert(fs.exists(filename) == true);
+
+let obj = fs.readJson(filename);
+assert(Json.stringify(obj) == Json.stringify(data));
```

Comment:
```suggestion
assert(obj == data);
```

I suspect this should work, right @hasanaburayyan ?

Reply:
I have attempted this approach, but it is not functioning as expected. I am unable to determine the underlying issue, even though the output of Json.stringify for these two objects is identical.

Reply:
Weirdly, I am unable to reproduce the phenomenon using the 'extern' mechanism. Regardless of how I modify the programming approach, the assertion always evaluates to true.

In the fs sdk, I utilize the 'as' operator to convert the result of JSON.parse into a Json object. This may not be considered as best practice? I have yet to discover how to create a Json object in JavaScript.


Here is the code.
json.main.w
```
class JsJson {  
    extern "./json.js" pub static writeJson(obj: Json);
    extern "./json.js" pub static readJson(): Json;
}

let data1 = Json {
    "foo": "bar", 
    "arr": [1, 2, 3, "test", { "foo": "bar" }]
};

JsJson.writeJson(data1);
let data2 = JsJson.readJson();

test "json compare" {
    log(Json.stringify(data1));
    log(Json.stringify(data2));
    assert(data1 == data2);
}
```

json.js
```javascript
const fs = require("fs");

exports.writeJson = function (obj) {
    fs.writeFileSync("data.json", JSON.stringify(obj, null, 2));
}

exports.readJson = function () {
    const content = fs.readFileSync("data.json");
    return JSON.parse(content);
};
```